### ISSUE TITLE
The consumer suddenly stops polling for new messages

### ISSUE THREAD

**Issue by Esya** [bbc/sqs-consumer#130]
Hi,

We're using sqs-consumer to poll our queue, but something strange happens; at times of activity, after a few minutes, the consumer just stops polling for new messages. If we restart our server, then it picks up the messages again. 

![2018-10-06-033945_1413x850_scrot](https://user-images.githubusercontent.com/937179/46566153-88208c80-c919-11e8-81c9-b1b5b6461129.png)

Here at 22:30 we restarted the server, and it started trying to receive again. Between 19:15 and 22:30 it did not pick any message from the queue. We're only specifying `queueUrl` - all other options are default. (Also there was no network outage or special event going on., this happens frequently)

Our queue is configured like so : 

![2018-10-06-014817_650x206_scrot](https://user-images.githubusercontent.com/937179/46564801-f8bfad00-c909-11e8-8781-3f0e659e3a57.png)

We're logging errors : 

```
 this._consumer.on("error", e => {
            logger.error(`Error with queue`, { queue: this._queueName, error: e.message });
 });
```

But this never triggers. We've also tried logging events, but when we stop receiving the messages, we do not get a single event fired from sqs-consumer.

Any ideas? Thanks!

**Comment by giles-t** [comment]
This suddenly starting happening to me today also. I haven't any changes to the process in months. Very strange.

**Comment by pschwyter** [comment]
We've been experiencing the same issue for weeks on `3.8.0`.

**Comment by hibearpanda** [comment]
Getting this issue as well. Did anyone try rolling back?

**Comment by Victordmdb** [comment]
Also having the same issue, did anyone find a solution?

**Comment by nspragg** [comment]
@Esya - what debugging measures have you taken so far? 

You mentioned that the `error` event never triggers. What about the `processing_error` event? Have you tried adding an unhandled exception handler?

**Comment by Esya** [comment]
@nspragg We did not get any unhandled exception, nor processing_error event. The queue just stalls for some reason. 

We've decided to rewrite it all using `aws-sdk` directly

**Comment by nspragg** [comment]
@Esya - You're welcome to send on a sample version of your consumer code to see if I can spot something. 

However, if you wish to continue with a different solution, you may consider other modules on npm like: https://github.com/TechnologyAdvice/Squiss

**Comment by Esya** [comment]
@nspragg We've just deployed two of our services using 4.0.0 - with no code changes on our end (But as mentioned previously, our consumer code is quite simple) - we'll see if this happens in the next 48 hours and I'll update this issue

Thanks!

**Comment by nspragg** [comment]
@Esya - Incidentally, the above questions are related to #136

**Comment by nspragg** [comment]
@Esya - It's also possible that AWS calls are hanging - we'll be investigating this in #137. 

Timeout updates are expected to be released this week.

**Comment by Esya** [comment]
@nspragg Yes, it does so for all instances of the sqs-consumer, after some time/some random amount of messages going through the queue 

We've implemented the following changes and we do not have any issues anymore : 

```diff
-        const consumerOptions: Consumer.Options = Object.assign(
-            {
-                queueUrl: `https://sqs.eu-west-1.amazonaws.com/[REDACTED]/${this._queueName}.fifo`,
-                region: "eu-west-1",
-                handleMessage: this.handler.bind(this)
-            },
-            options
-        );
-
-
-        this._consumer = Consumer.create(consumerOptions);
-        this._consumer.on("error", e => {
-            logger.error(`Error with queue`, { queue: this._queueName, error: e.message });
+        this._poller = new Squiss({
+            awsConfig: {region: "eu-west-1"},
+            queueUrl: `https://sqs.eu-west-1.amazonaws.com/[REDACTED]/${this._queueName}.fifo`,
+            maxInFlight: 1
+        });
+
+        this._poller.on("message", msg => {
+            this.handler(msg);
         });
```

**Comment by nspragg** [comment]
Fix released in v4.1.0
